### PR TITLE
chore/cli: refactor base-path

### PR DESCRIPTION
**Pull request by kanishkatn**
## Changes

- Separate path for data and config

## Issues

#3211

## Primary Reviewer

@EclesioMeloJunior

### CODE REVIEW

File: chain/kusama/defaults.go [hunk 1]
```diff
@@ -5,23 +5,27 @@ package kusama
...
 
 var (
 	// defaultName Default node name
 	defaultName = "Kusama"
 	// defaultID Default chain ID
 	defaultID = "ksmcc3"
-	// defaultBasePath Default node base directory path
-	defaultBasePath = "~/.gossamer/kusama"
+	// defaultConfigDir is the default config directory path
+	defaultConfigDir = xdg.ConfigHome + "/kusama/config"
```
Comment (q9f) [inline chain/kusama/defaults.go:17]:
```suggestion
	defaultConfigDir = xdg.ConfigHome + "/gossamer/kusama"
```

File: chain/kusama/defaults.go [hunk 2]
```diff
@@ -5,23 +5,27 @@ package kusama
...
+	// defaultDataPath is the default data directory path
+	defaultDataPath = xdg.DataHome + "/kusama/data"
```
Comment (q9f) [inline chain/kusama/defaults.go:19]:
```suggestion
	defaultDataPath = xdg.DataHome + "/gossamer/kusama"
```

File: chain/paseo/defaults.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	"github.com/adrg/xdg"
+)
+
+var (
+	// defaultName Default node name
+	defaultName = "Paseo"
+	// defaultID Default chain ID
+	defaultID = "paseo"
+	// defaultConfigDir is the default config directory path
+	defaultConfigDir = xdg.ConfigHome + "/paseo/config"
```
Comment (q9f) [inline chain/paseo/defaults.go:17]:
```suggestion
	defaultConfigDir = xdg.ConfigHome + "/gossamer/paseo"
```

File: chain/paseo/defaults.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	// defaultDataPath is the default data directory path
+	defaultDataPath = xdg.DataHome + "/paseo/data"
```
Comment (q9f) [inline chain/paseo/defaults.go:19]:
```suggestion
	defaultDataPath = xdg.DataHome + "/gossamer/paseo"
```

File: chain/polkadot/defaults.go [hunk 5]
```diff
@@ -5,23 +5,27 @@ package polkadot
...
 
 var (
 	// defaultName Default node name
 	defaultName = "Polkadot"
 	// defaultID Default chain ID
 	defaultID = "polkadot"
-	// defaultBasePath Default node base directory path
-	defaultBasePath = "~/.gossamer/polkadot"
+	// defaultConfigDir is the default config directory path
+	defaultConfigDir = xdg.ConfigHome + "/polkadot/config"
```
Comment (q9f) [inline chain/polkadot/defaults.go:17]:
```suggestion
	defaultConfigDir = xdg.ConfigHome + "/gossamer/polkadot"
```

File: chain/polkadot/defaults.go [hunk 6]
```diff
@@ -5,23 +5,27 @@ package polkadot
...
+	// defaultDataPath is the default data directory path
+	defaultDataPath = xdg.DataHome + "/polkadot/data"
```
Comment (q9f) [inline chain/polkadot/defaults.go:19]:
```suggestion
	defaultDataPath = xdg.DataHome + "/gossamer/polkadot"
```

File: chain/westend-dev/defaults.go [hunk 7]
```diff
@@ -5,23 +5,27 @@ package westenddev
...
-const (
+var (
 	// defaultName is the default node name
 	defaultName = "Westend"
 	// defaultID is the default node ID
 	defaultID = "westend_dev"
-	// defaultBasePath is the default basepath for the westend dev node
-	defaultBasePath = "~/.gossamer/westend-dev"
+	// defaultConfigDir is the default config directory path
+	defaultConfigDir = xdg.ConfigHome + "/westend-dev/config"
```
Comment (q9f) [inline chain/westend-dev/defaults.go:17]:
```suggestion
	defaultConfigDir = xdg.ConfigHome + "/gossamer/westend-dev"
```

File: chain/westend-dev/defaults.go [hunk 8]
```diff
@@ -5,23 +5,27 @@ package westenddev
...
+	// defaultDataPath is the default data directory path
+	defaultDataPath = xdg.DataHome + "/westend-dev/data"
```
Comment (q9f) [inline chain/westend-dev/defaults.go:19]:
```suggestion
	defaultDataPath = xdg.DataHome + "/gossamer/westend-dev"
```

File: chain/westend-local/default.go [hunk 9]
```diff
@@ -41,7 +35,8 @@ func DefaultConfig() *cfg.Config {
 // DefaultAliceConfig returns a westend local node configuration
 func DefaultAliceConfig() *cfg.Config {
 	config := DefaultConfig()
-	config.BasePath = DefaultBasePathAlice
+	config.ConfigDir = xdg.ConfigHome + "/westend-local-alice/config"
```
Comment (q9f) [inline chain/westend-local/default.go:38]:
```suggestion
	config.ConfigDir = xdg.ConfigHome + "/gossamer/westend-local/alice"
```

File: chain/westend-local/default.go [hunk 10]
```diff
@@ -41,7 +35,8 @@ func DefaultConfig() *cfg.Config {
...
+	config.DataDir = xdg.DataHome + "/westend-local-alice/data"
```
Comment (q9f) [inline chain/westend-local/default.go:39]:
```suggestion
	config.DataDir = xdg.DataHome + "/gossamer/westend-local/alice"
```

File: chain/westend-local/default.go [hunk 11]
```diff
@@ -54,7 +49,8 @@ func DefaultAliceConfig() *cfg.Config {
 // DefaultBobConfig returns a westend local node configuration with bob as the authority
 func DefaultBobConfig() *cfg.Config {
 	config := DefaultConfig()
-	config.BasePath = DefaultBasePathBob
+	config.ConfigDir = xdg.ConfigHome + "/westend-local-bob/config"
```
Comment (q9f) [inline chain/westend-local/default.go:52]:
```suggestion
	config.ConfigDir = xdg.ConfigHome + "/gossamer/westend-local/bob"
```

File: chain/westend-local/default.go [hunk 12]
```diff
@@ -54,7 +49,8 @@ func DefaultAliceConfig() *cfg.Config {
...
+	config.DataDir = xdg.DataHome + "/westend-local-bob/data"
```
Comment (q9f) [inline chain/westend-local/default.go:53]:
```suggestion
	config.DataDir = xdg.DataHome + "/gossamer/westend-local/bob"
```

File: chain/westend-local/default.go [hunk 13]
```diff
@@ -67,7 +63,8 @@ func DefaultBobConfig() *cfg.Config {
 // DefaultCharlieConfig returns a westend local node configuration with charlie as the authority
 func DefaultCharlieConfig() *cfg.Config {
 	config := DefaultConfig()
-	config.BasePath = DefaultBasePathCharlie
+	config.ConfigDir = xdg.ConfigHome + "/westend-local-charlie/config"
```
Comment (q9f) [inline chain/westend-local/default.go:66]:
```suggestion
	config.ConfigDir = xdg.ConfigHome + "/gossamer/westend-local-charlie"
```

File: chain/westend-local/default.go [hunk 14]
```diff
@@ -67,7 +63,8 @@ func DefaultBobConfig() *cfg.Config {
...
+	config.DataDir = xdg.DataHome + "/westend-local-charlie/data"
```
Comment (q9f) [inline chain/westend-local/default.go:67]:
```suggestion
	config.DataDir = xdg.DataHome + "/gossamer/westend-local/charlie"
```

File: chain/westend/defaults.go [hunk 15]
```diff
@@ -5,23 +5,27 @@ package westend
...
 
 var (
 	// defaultName Default node name
 	defaultName = "Westend"
 	// defaultID Default chain ID
 	defaultID = "westend2"
-	// defaultBasePath Default node base directory path
-	defaultBasePath = "~/.gossamer/westend"
+	// defaultConfigDir is the default config directory path
+	defaultConfigDir = xdg.ConfigHome + "/westend/config"
```
Comment (q9f) [inline chain/westend/defaults.go:17]:
```suggestion
	defaultConfigDir = xdg.ConfigHome + "/gossamer/westend"
```

File: chain/westend/defaults.go [hunk 16]
```diff
@@ -5,23 +5,27 @@ package westend
...
+	// defaultDataPath is the default data directory path
+	defaultDataPath = xdg.DataHome + "/westend/data"
```
Comment (q9f) [inline chain/westend/defaults.go:19]:
```suggestion
	defaultDataPath = xdg.DataHome + "/gossamer/westend"
```

File: cmd/gossamer/commands/root.go [hunk 17]
```diff
@@ -580,15 +576,15 @@ func execRoot(cmd *cobra.Command) error {
 	}
 
 	// load user keys if specified
-	if err := unlockKeystore(ks.Acco, config.BasePath, config.Account.Unlock, password); err != nil {
+	if err := unlockKeystore(ks.Acco, config.DataDir, config.Account.Unlock, password); err != nil {
```
Comment (q9f) [inline cmd/gossamer/commands/root.go:579]:
```suggestion
	if err := unlockKeystore(ks.Acco, config.ConfigDir, config.Account.Unlock, password); err != nil {
```

File: cmd/gossamer/commands/root.go [hunk 18]
```diff
@@ -580,15 +576,15 @@ func execRoot(cmd *cobra.Command) error {
...
 		return fmt.Errorf("failed to unlock keystore: %s", err)
 	}
 
-	if err := unlockKeystore(ks.Babe, config.BasePath, config.Account.Unlock, password); err != nil {
+	if err := unlockKeystore(ks.Babe, config.DataDir, config.Account.Unlock, password); err != nil {
```
Comment (q9f) [inline cmd/gossamer/commands/root.go:583]:
```suggestion
	if err := unlockKeystore(ks.Babe, config.ConfigDir, config.Account.Unlock, password); err != nil {
```

File: cmd/gossamer/commands/root.go [hunk 19]
```diff
@@ -580,15 +576,15 @@ func execRoot(cmd *cobra.Command) error {
...
 		return fmt.Errorf("failed to unlock keystore: %s", err)
 	}
 
-	if err := unlockKeystore(ks.Gran, config.BasePath, config.Account.Unlock, password); err != nil {
+	if err := unlockKeystore(ks.Gran, config.DataDir, config.Account.Unlock, password); err != nil {
```
Comment (q9f) [inline cmd/gossamer/commands/root.go:587]:
```suggestion
	if err := unlockKeystore(ks.Gran, config.ConfigDir, config.Account.Unlock, password); err != nil {
```

File: cmd/gossamer/commands/utils.go [hunk 20]
```diff
@@ -14,6 +14,8 @@ import (
 	"syscall"
 	"time"
 
+	"github.com/ChainSafe/gossamer/chain/paseo"
```
Comment (q9f) [inline cmd/gossamer/commands/utils.go:17]: let's split these two features into separate PRs

File: config/config.go [hunk 21]
```diff
@@ -126,9 +133,11 @@ func (cfg *Config) ValidateBasic() error {
 
 // BaseConfig is to marshal/unmarshal toml global config vars
 type BaseConfig struct {
+	ConfigDir string
+
 	Name               string                      `mapstructure:"name,omitempty"`
 	ID                 string                      `mapstructure:"id,omitempty"`
-	BasePath           string                      `mapstructure:"base-path,omitempty"`
+	DataDir            string                      `mapstructure:"data-dir,omitempty"`
```
Comment (q9f) [inline config/config.go:140]:
```suggestion
	ConfigDir          string                      `mapstructure:"config-dir,omitempty"`
	DataDir            string                      `mapstructure:"data-dir,omitempty"`
```

**Review by q9f — COMMENTED** [review]
Left some comments but did not review everything. Make sure:
- [ ] not to mix this with the Paseo PR
- [ ] use `/gossamer` as first child after the respective XDG dir
- [ ] omit the trailing `/config` or `/data`
- [ ] make sure `dataDir` only has data (db, chain, state, peers) - nothing else
- [ ] make sure all keys (network key, account keys) are in `configDir`

File: cmd/gossamer/commands/root.go (hunk 17 above)
Comment (kanishkatn) [reply cmd/gossamer/commands/root.go:579]: I consider keys as data and not config. Could you tell me why you think otherwise?

Comment (q9f) [reply cmd/gossamer/commands/root.go:579]: the key difference is what you do with it:
- disk full? wipe data, retain keys & config
- borked database? wipe data, retain keys & config

while your point is reasonable, from a user perspective, there is no reason to have keys in a directory that might be subject to regular purges

File: dot/services.go [hunk 22]
```diff
@@ -313,7 +313,7 @@ func (nodeBuilder) createNetworkService(config *cfg.Config, stateSrvc *state.Ser
 	networkConfig := network.Config{
 		LogLvl:            networkLogLevel,
 		BlockState:        stateSrvc.Block,
-		BasePath:          config.BasePath,
+		BasePath:          config.ConfigDir,
```
Comment (q9f) [inline dot/services.go:316]:
```suggestion
		ConfigDir:          config.ConfigDir,
```

**Comment by q9f** [comment]
#3698
